files/external_node_v2.rb

```diff
@@ -213,8 +213,14 @@ def enc(certname)
   end
   res = http.start { |http| http.request(req) }
 
-  raise "Error retrieving node #{certname}: #{res.class}\nCheck Foreman's /var/log/foreman/production.log for more information." unless res.code == "200"
-  res.body
+  return res.body if res.code == '200'
+
+  if (500..599).include?(res.code.to_i)
+    raise Net::HTTPServerException.new('500 error in Foreman', res)
```

Comment:
1) "500" isn't correct for a non-500 error (e.g. 503), it should use the code.

<pre>
# node.rb $(hostname -f)
2016-12-22 05:51:30 56acff41 [app] [I] Started GET "/node/foreman.example.com?format=yml" for 127.0.0.1 at 2016-12-22 05:51:30 -0500
2016-12-22 05:51:30 56acff41 [app] [I] Processing by HostsController#externalNodes as YML
2016-12-22 05:51:30 56acff41 [app] [I]   Parameters: {"name"=>"foreman.example.com"}
2016-12-22 05:51:30 56acff41 [app] [I] Completed 503 Service Unavailable in 4ms (ActiveRecord: 0.8ms)
500 error in Foreman
</pre>

2) This doesn't seem to show any logging to stderr?